Plugins/Encounter/Encounter.cpp

```diff
@@ -278,4 +281,95 @@ ArgumentStack Encounter::GetCurrentNumSpawned(ArgumentStack&& args)
     return Services::Events::Arguments(retVal);
 }
 
+ArgumentStack Encounter::GetEncounterGeometry(ArgumentStack&& args)
+{
+    std::string retVal;
+
+    if (auto *pObject = encounter(args))
+    {
+        if (auto *pEncounter = Utils::AsNWSEncounter(pObject))
```

Comment:
`encounter(args)` already returns a `CNWSEncounter`, this second if is not needed. Just remove this one, and rename `pObject` to `pEncounter`. Same in the other function.